tests/console/zypper_lifecycle.pm

```diff
@@ -143,7 +154,9 @@ sub run {
         mv /var/lib/lifecycle/data/$prod.lifecycle.orig /var/lib/lifecycle/data/$prod.lifecycle
     fi";
     #
-    select_console 'user-console';
+    if ((check_var('ARCH', 'aarch64') || (check_var('ARCH', 'x86_64')) {
+        select_console 'user-console';
+    }
```

Comment:
Better make it more obvious why these steps are taken, e.g. set a variable `$need_root_console` depending on the bug conditions and then call here:

```perl
select_console 'user-console' unless $needs_root_console:
```

As a general hint: Always think about how a junior person in 5 years would read the code and ask yourself if that person can understand the code and why it is like it is :)

Reply:
It's not about junior or something, it's code clear and simple. period.

Reply:
Sorry, not sure what you mean by that.

Reply:
I've changed the code as you suggested, please review it again for this.

Reply:
good. Please add a comment above the "if ((check_var('ARCH', 'aarch64') || (check_var('ARCH', 'x86_64'))" why this is needed (for the workaround).

Reply:
Changed with comment.  please check again to see which part need change. thanks.